Docs alerting: meta monitoring topics (#74440)
grafana/grafana: docs/sources/alerting/monitor/meta-monitoring/_index.md

aliases:

>   - meta-monitoring/

Reviewer: I'm not sure this alias is doing anything. I am pretty sure it would have been redirecting https://grafana.com/docs/grafana/latest/alerting/set-up/meta-monitoring/ to itself.

I think it can be removed unless it was actually meant to redirect something like https://grafana.com/docs/grafana/latest/alerting/meta-monitoring/ to this page?

Author: done!

>   - alerting/meta-monitoring/

Reviewer: This currently redirects https://grafana.com/docs/grafana/latest/alerting/set-up/alerting/meta-monitoring/ to https://grafana.com/docs/grafana/latest/alerting/set-up/meta-monitoring/.

Keeping the same alias will mean that  https://grafana.com/docs/grafana/latest/alerting/set-up/alerting/meta-monitoring/ will now redirect to your new page location https://grafana.com/docs/grafana/latest/alerting/monitor/meta-monitoring/.

If the original redirect was correct and desirable, then I believe this alias is still correct but I wanted to double check that the original redirect is correct.

Author: this is what i want to happen: that this page https://grafana.com/docs/grafana/latest/alerting/set-up/alerting/meta-monitoring/ redirects to your new page location https://grafana.com/docs/grafana/latest/alerting/monitor/meta-monitoring/.

-> what do i need to change?

Reviewer: Cool, I believe that this alias is still correct and will redirect https://grafana.com/docs/grafana/latest/alerting/set-up/alerting/meta-monitoring/ to https://grafana.com/docs/grafana/latest/alerting/monitor/meta-monitoring/ once this PR is merged and backported.

Reviewer: You can check locally by running `make docs` and browsing to http://localhost:3002/docs/grafana/latest/alerting/set-up/alerting/meta-monitoring/. You should be redirected to your new page. If it 404s, the alias is incorrect.

Author: cool, thanks

>   - /alerting/set-up/meta-monitoring/

Reviewer: I believe the intent of this alias is to redirect the old location of https://grafana.com/docs/grafana/latest/alerting/set-up/meta-monitoring/ to https://grafana.com/docs/grafana/latest/alerting/monitor/meta-monitoring/.

The alias to do that is:

```suggestion
  - ../set-up/meta-monitoring/
```

  products:
    - enterprise
    - oss
    - cloud
title: Meta monitoring
weight: 100
---

# Meta monitoring

Meta monitoring is the process of monitoring your monitoring system and alerting when your monitoring is not working as it should.

Whether you use Grafana-managed alerts or Grafana Mimir-managed alerts, meta monitoring is possible both on-premise and in Grafana Cloud.

In order to enable you to meta monitor, Grafana provides predefined metrics for both on-premise and Cloud.

Identify which metrics are critical to your monitoring system (i.e. Grafana) and then set up how you want to monitor them.

## Metrics for Grafana-managed alerts

{{% admonition type="note" %}}
These metrics are not available for Grafana Cloud.

> There are no metrics in the `grafanacloud-usage` data source that can be used to meta monitor Grafana-managed alerts.

Reviewer 2: ```suggestion
There are no metrics in the `grafanacloud-usage` data source at this time that can be used to meta monitor Grafana-managed alerts.
```

> These metrics are not available at present in Grafana Cloud.

> ## Metrics for Grafana Mimir

Reviewer 3: ```suggestion
## Metrics for Mimir-managed alerts
```

{{% admonition type="note" %}}

> This metric is available in OSS, on-premise, and Grafana Cloud.

Reviewer 2: ```suggestion
These metrics are available in OSS, on-premise, and Grafana Cloud.
```

Reviewer 2: There is more than one metric in total, although for some reason I documented just one of them when I wrote this page. Not sure if we should change it from singular to plural in that case?

{{% /admonition %}}

> To meta monitor in Alertmanager, you need a Prometheus/Mimir server, or another metrics database to collect and store metrics exported by the Mimir ruler.

Reviewer 3: ```suggestion
To meta monitor Grafana Mimir-managed alerts, open source and on-premise users need a Prometheus/Mimir server, or another metrics database to collect and store metrics exported by the Mimir ruler.
```

Reviewer 3: Add something like 

Grafana Cloud customers can find this metric in their `grafanacloud-usage` datasource.

Reviewer 3: note that its going to have a slightly different name though so that might cause confusion. In Grafana Cloud its:
`grafanacloud_instance_rule_evaluation_failures_total:rate5m{}`

#### rule_evaluation_failures_total

This metric is a counter that shows you the total number of rule evaluation failures.

## Metrics for Alertmanager

{{% admonition type="note" %}}
These metrics are available for OSS, on-premise, and some are available in Grafana Cloud.

Use the data source and Metrics browser in Grafana Cloud called `grafanacloud-usage` that is provisioned for all Grafana Cloud customers to view all available meta monitoring and usage metrics that are available in Grafana Cloud.
{{% /admonition %}}

> To meta monitor in Alertmanager, you need a Prometheus/Mimir server, or another metrics database to collect and store metrics exported by Alertmanager.

Reviewer 3: ```suggestion
To meta monitor the Alertmanager, you need a Prometheus/Mimir server, or another metrics database to collect and store metrics exported by Alertmanager.
```

Reviewer 3: note that this is only true for oss or on-prem Enterprise customers. If you're cloud, we provide that database for you. :)

{{% admonition type="note" %}}
For Grafana Cloud only.
{{% /admonition %}}

Use insights logs to help you determine which alerting and recording rules are failing to evaluate and why. These logs contain helpful information on specific alert rules that are failing, provide you with the actual error message, and help you evaluate what is going wrong.

### Before you begin

To view your insights logs, you must have the following:

 - A Grafana Cloud account
 - Admin or Editor user permissions for the managed Grafana Cloud instance

### Procedure

To explore logs pertaining to failing alerting and recording rules, complete the following steps.

1. Log on to your instance and click the      **Explore** (compass) icon in the menu sidebar.
1. Use the data sources dropdown located at the top of the page to select the data source.
The data source name should be similar to `grafanacloud-<yourstackname>-usage-insights`.
1. To find the logs you want to see, use the **Label filters** and **Line contains** options in the query editor.

To look at a particular stack, you can filter by **instance_id** instead of **org_id**.

The following is an example query that would surface insights logs:

> {org_id="<your-org-id>"} | logfmt | component = `ruler` | msg = `Evaluating rule failed`

Reviewer 3: would it make sense to put this in a dedicated code block?

1. Click **Run query**.

>    In the **Logs** section, view specific information on which alert rule is failing and why.

Reviewer 3: Do we want a list with just 1 number in it? Or should this be step 2 and "You can see" be step 3?

Author: done

>    You can see the rule contents ("rule"), the rule name ("name"), the name of the group it’s in ("group"), and the error message ("err").

Reviewer 3: Do you think it would be helpful to have a screenshot here?
If so, here's a link to one: https://drive.google.com/file/d/1wuINfWpQk175SCnQrlAcUAiGlR42pYXa/view?usp=sharing

>    You can see the rule contents ("rule"), the rule name ("name"), the name of the group it’s in ("group"), and the error message ("err").

Reviewer 3: ```suggestion
   You can see the rule contents (in the `rule` field), the rule name (in the `name` field), the name of the group it’s in (in the `group` field), and the error message (in the `err` field).
```
